### PR TITLE
feat(sumo): upgrade to SUMO 1.17.0

### PR DESCRIPTION
**Pull request by kschrab**
## Description

Upgrades SUMO coupling interface to use latest version SUMO.

## Issue(s) related to this PR

* Resolves internal issue 608
  
## Affected parts of the online documentation

Changes in the documentation required?

No

## Definition of Done

**Prerequisites**

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] Your GitHub user id is linked with your Eclipse Account.

**Required**

- [x] The title of this merge request follows the scheme `type(scope): description`  (in the style of [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary))
- [x] You have assigned a suitable label to this pull request (e.g., `enhancement`, or `bugfix`)
- [x] `origin/main` has been merged into your Fork.
- [x] Coding guidelines have been followed (see CONTRIBUTING.md).
- [x] All checks on GitHub pass.
- [x] All tests on [Jenkins](https://ci.eclipse.org/mosaic/job/mosaic/) pass.

**Requested** (can be enforced by maintainers)

- [ ] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [ ] If a bug has been fixed, a new unit test has been written (beforehand) to prove misbehavior
- [x] There are no new SpotBugs warnings.

### CODE REVIEW

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/LibSumoAmbassador.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
  */
 public class LibSumoAmbassador extends SumoAmbassador {
 
-    private static final String VALID_LIBSUMO_VERSIONS = "1\\.1[456]\\.";
+    private static final String VALID_LIBSUMO_VERSIONS = "1\\.1[67]\\.";
```
Comment (schwepmo) [inline fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/LibSumoAmbassador.java:37]: What happens if someone wants to use MOSAIC using SUMO 1.15.0 or lower together with libsumo?

Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/LibSumoAmbassador.java:37]: Error message :) I don't think its necessary to support the older versions.

Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/LibSumoAmbassador.java:37]: But speaking of Libsumo, I should test it at least locally with 1.17.0, so please don't merge yet.

Comment (schwepmo) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/LibSumoAmbassador.java:37]: I just assumed you would've done that. 😛 Let me know when this is ready to merge.

Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/LibSumoAmbassador.java:37]: Done, everything fine :)

**Pull request merged**